### PR TITLE
bazel: update to 3.1.0

### PR DESCRIPTION
**Pull request by junr03**
Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

**Comment by junr03** [comment]
Taking a look at what it would mean to update to 3.1.0 to avoid the linking issue with bazelisk. I understand this is sweeping the underlying problem under the rug by complying with what is installed in the CI machines, but I'd rather unblock CI and then figure out how to correctly install and link bazelisk in macos.

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
       - run: ls -lh dist/
       - name: start simulator
         run: ./ci/mac_start_emulator.sh
-      - run: bazel mobile-install --fat_apk_cpu=x86 --start_app //examples/kotlin/hello_world:hello_envoy_kt
+      - run: bazel mobile-install --host_force_python=PY2 --fat_apk_cpu=x86 //examples/kotlin/hello_world:hello_envoy_kt
```
Comment (buildbreaker) [inline .github/workflows/android.yml:92]:
```suggestion
      - run: bazel mobile-install --host_force_python=PY2 --fat_apk_cpu=x86 //examples/kotlin/hello_world:hello_envoy_kt
```
https://github.com/bazelbuild/bazel/issues/7899

Comment (junr03) [reply .github/workflows/android.yml:92]: No, that is just a red herring they already updated to pyhon 3 in https://github.com/bazelbuild/bazel/commit/d93a146463e20f4eedc5ae658b03ee0260750aa7#diff-947a45ac6fb178dbfb7cc530fddd6b9a

**Comment by junr03** [comment]
closing in favor of https://github.com/lyft/envoy-mobile/pull/854

**Pull request closed without merging**